qiita_db/portal.py

```diff
@@ -26,38 +162,199 @@ def get_studies_by_portal(portal):
             All study ids in the database that match the given portal
         """
         conn_handler = SQLConnectionHandler()
-        portal_id = convert_to_id(portal, 'portal_type', 'portal')
         sql = """SELECT study_id FROM qiita.study_portal
                  WHERE portal_type_id = %s"""
         return {x[0] for x in
-                conn_handler.execute_fetchall(sql, [portal_id])}
+                conn_handler.execute_fetchall(sql, [self._id])}
+
+    def _check_studies(self, studies):
+        conn_handler = SQLConnectionHandler()
+        # Check if any study IDs given do not exist.
+        sql = "SELECT study_id from qiita.study WHERE study_id IN %s"
+        existing = [x[0] for x in conn_handler.execute_fetchall(
+            sql, [tuple(studies)])]
+        if len(existing) != len(studies):
+            bad = map(str, (set(studies).difference(existing)))
+            raise QiitaDBError("The following studies do not exist: %s" %
+                               ", ".join(bad))
 
-    def add_study_to_portal(study, portal):
-        """Adds study to given portal
+    def add_studies(self, studies):
+        """Adds studies to given portal
 
         Parameters
         ----------
-        portal : str
-            Portal to associate with.
+        studies : list of int
```

Comment:
people have taken `list of int` to mean any iterator of ints in the past. I wanted to be explicit with the tuple because then the function will work with any iterator, since psycopg2 is the one requiring a tuple.